<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>900</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Microsoft YaHei Light</family>
    <pointsize>-1</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <property name="styleSheet">
   <string notr="true">*{
   font-family:&quot;Microsoft YaHei Light&quot;;
   font-size:18px;
}</string>
  </property>
  <widget class="QFrame" name="frame">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>1180</width>
     <height>880</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Microsoft YaHei Light</family>
     <pointsize>-1</pointsize>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::ClickFocus</enum>
   </property>
   <property name="styleSheet">
    <string notr="true">#frame{
    background-color: transparent;
    border-image: url(:/Images/bg.jpg);
    border-radius:6px;
    opacity:0.5;
}</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QLabel" name="Icon">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>5</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border-image:url(:/Images/icon.png);</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="Title">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>0</y>
      <width>751</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Microsoft YaHei UI Light</family>
      <pointsize>20</pointsize>
      <weight>3</weight>
      <italic>false</italic>
      <bold>false</bold>
      <stylestrategy>PreferAntialias</stylestrategy>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 25 20pt &quot;Microsoft YaHei UI Light&quot;;</string>
    </property>
    <property name="text">
     <string>数独运算器</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="MinButton">
    <property name="geometry">
     <rect>
      <x>1087</x>
      <y>0</y>
      <width>48</width>
      <height>48</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="styleSheet">
     <string notr="true">#MinButton{
   border-image:url(:/Images/min.ico);
   border-radius:5px;
}
#MinButton:pressed{
   border-image:url(:/Images/min_press.bmp);
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="CloseButton">
    <property name="geometry">
     <rect>
      <x>1132</x>
      <y>0</y>
      <width>48</width>
      <height>48</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="baseSize">
     <size>
      <width>0</width>
      <height>0</height>
     </size>
    </property>
    <property name="styleSheet">
     <string notr="true">#CloseButton{
   border-image:url(:/Images/close.ico);
   border-radius:5px;
}

#CloseButton:pressed{
   border-image:url(:/Images/close_press.ico);
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QTableWidget" name="Grid">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>90</y>
      <width>765</width>
      <height>765</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:rgb(239, 239, 239);
font: 75 40pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <property name="autoScroll">
     <bool>false</bool>
    </property>
    <property name="editTriggers">
     <set>QAbstractItemView::AnyKeyPressed|QAbstractItemView::DoubleClicked</set>
    </property>
    <property name="showDropIndicator" stdset="0">
     <bool>true</bool>
    </property>
    <property name="dragDropOverwriteMode">
     <bool>true</bool>
    </property>
    <property name="textElideMode">
     <enum>Qt::ElideLeft</enum>
    </property>
    <property name="gridStyle">
     <enum>Qt::SolidLine</enum>
    </property>
    <property name="cornerButtonEnabled">
     <bool>false</bool>
    </property>
    <property name="rowCount">
     <number>9</number>
    </property>
    <property name="columnCount">
     <number>9</number>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderMinimumSectionSize">
     <number>85</number>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>85</number>
    </attribute>
    <attribute name="horizontalHeaderHighlightSections">
     <bool>false</bool>
    </attribute>
    <attribute name="verticalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <attribute name="verticalHeaderMinimumSectionSize">
     <number>85</number>
    </attribute>
    <attribute name="verticalHeaderDefaultSectionSize">
     <number>85</number>
    </attribute>
    <attribute name="verticalHeaderHighlightSections">
     <bool>false</bool>
    </attribute>
    <row>
     <property name="text">
      <string>新建行</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>新建行</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>新建行</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>新建行</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>新建行</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>新建行</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>新建行</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>新建行</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>新建行</string>
     </property>
    </row>
    <column>
     <property name="text">
      <string>新建列</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>新建列</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>新建列</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>新建列</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>新建列</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>新建列</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>新建列</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>新建列</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>新建列</string>
     </property>
    </column>
    <item row="0" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="0" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="0" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="0" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="0" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="0" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="0" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="0" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="0" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="1" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="2" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="3" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="4" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="5" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="6" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="7" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="0">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="1">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="2">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="3">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="4">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="5">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="6">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="7">
     <property name="text">
      <string/>
     </property>
    </item>
    <item row="8" column="8">
     <property name="text">
      <string/>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="Start">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>270</y>
      <width>201</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 14pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <property name="text">
     <string>开始计算数独</string>
    </property>
   </widget>
   <widget class="QLabel" name="StateLable">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>60</y>
      <width>341</width>
      <height>71</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 18pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <property name="text">
     <string>状态：等待计算</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QComboBox" name="ShowSteps">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>130</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 16pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <item>
     <property name="text">
      <string>是</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>否</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="ShowStepLable">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>130</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 14pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <property name="text">
     <string>是否显示过程：</string>
    </property>
   </widget>
   <widget class="QLabel" name="Description">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>480</y>
      <width>341</width>
      <height>371</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font:  15pt &quot;楷体&quot;;</string>
    </property>
    <property name="text">
     <string>说明：可以单击左边的格子并按键盘输入数字1-9将需要解的数独填入表格中，留下空白表示待解决的数字，按&quot;开始计算数独&quot;即可计算。

特别提示：对于不合法的字符和1-9以外的数字将被视作空白即待解决的数字。</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="ClearGrid">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>270</y>
      <width>121</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 12pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <property name="text">
     <string>清空表格</string>
    </property>
   </widget>
   <widget class="QPushButton" name="FileIn">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>370</y>
      <width>321</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 12pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <property name="text">
     <string>从文件中读入数独</string>
    </property>
   </widget>
   <widget class="QPushButton" name="FileOut">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>420</y>
      <width>321</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 12pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <property name="text">
     <string>将结果保存到文件</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Random">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>320</y>
      <width>321</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 12pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <property name="text">
     <string>随机生成一个数独谜题</string>
    </property>
   </widget>
   <widget class="QComboBox" name="Algorithm">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>220</y>
      <width>181</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 12pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <item>
     <property name="text">
      <string>舞蹈链(较快)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>深度优先搜索</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="AlgorithmLable">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>220</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 14pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <property name="text">
     <string>解题算法：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="StepsSpeed">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>170</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 16pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <item>
     <property name="text">
      <string>快</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>中</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>慢</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="StepsSpeedLable">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>170</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 14pt &quot;Microsoft YaHei UI&quot;;</string>
    </property>
    <property name="text">
     <string>过程显示速度:</string>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
